adam/perception/_matcher.py

```diff
@@ -248,6 +250,10 @@ def match(self, *, debug: bool = False):
         we yield the mapping.
 
         """
+        # Check rendering debug flag to see if we should render the graph
+        if debug_function is not None:
```

Comment:
This can probably be hidden under the `if debug` conditional below so that we are spared the functional call overhead (significant in Python :-( ) when debug mode is not enabled.